### PR TITLE
Fix retention policy in influxdb sink

### PR DESCRIPTION
**Pull request by ringtail**
Link to #2086 .  The previous pr can not pass unit test because of some unknown reasons.

### CODE REVIEW

File: metrics/sinks/influxdb/influxdb.go [hunk 1]
```diff
@@ -251,23 +251,57 @@ func (sink *influxdbSink) ensureClient() error {
 
 func (sink *influxdbSink) createDatabase() error {
 	if err := sink.ensureClient(); err != nil {
+		glog.Errorf("Failed to create influxdb client:%s", err.Error())
```
Comment (andyxning) [inline metrics/sinks/influxdb/influxdb.go]: `client:%s` -> `client: %s`

File: metrics/sinks/influxdb/influxdb.go [hunk 2]
```diff
@@ -251,23 +251,57 @@ func (sink *influxdbSink) ensureClient() error {
...
 			}
 		}
+
+		q = influxdb.Query{
+			Command: fmt.Sprintf(`CREATE DATABASE %s`, sink.c.DbName),
+		}
+
+		_, err := sink.client.Query(q)
+		if err != nil {
+			glog.Errorf("Failed to create database in influxdb:%s", err.Error())
```
Comment (andyxning) [inline metrics/sinks/influxdb/influxdb.go]: `influxdb:%s` -> `influxdb: %s`

**Comment by andyxning** [comment]
@ringtail Sorry for the late. You need to also run `make fmt` because `make sanitize` is reporting error.

**Comment by ringtail** [comment]
> @ringtail Sorry for the late. You need to also run `make fmt` because `make sanitize` is reporting error.

yes,that's really strange.I have run `make fmt` and `make sanitize` locally. but there is nothing warning.

```
➜  heapster git:(feature/fix-comments) make fmt
find . -type f -name "*.go" | grep -v "./vendor*" | xargs gofmt -s -w
➜  heapster git:(feature/fix-comments) ✗ make sanitize
hooks/check_boilerplate.sh
hooks/check_gofmt.sh
hooks/run_vet.sh
➜  heapster git:(feature/fix-comments) ✗
```

**Pull request closed without merging**